src/subscription_manager/cli_command/environments.py

```diff
@@ -77,3 +91,82 @@ def _do_command(self):
             system_exit(os.EX_SOFTWARE, mapped_message)
         except Exception as e:
             handle_exception(_("Error: Unable to retrieve environment list from server"), e)
+
+    def _set_environments(self):
+        if self.cp.has_capability(MULTI_ENV):
+            if not self.identity.is_valid():
+                system_exit(ERR_NOT_REGISTERED_CODE, ERR_NOT_REGISTERED_MSG)
+            self.cp.updateConsumer(
+                self.identity.uuid,
+                environments=self._process_environments(
+                    self.cp,
+                    self.cp.getOwner(self.identity.uuid)['key'],
+                    self.options
+                )
+            )
+            print(_("Environments updated."))
+        else:
+            system_exit(os.EX_UNAVAILABLE, _("Error: Server does not support environment updates."))
+
+    def _list_environments(self):
+        if not self.cp.has_capability(MULTI_ENV) and (self.options.enabled or self.options.disabled):
+            system_exit(os.EX_UNAVAILABLE, _("Error: Server does not support multi-environment operations."))
+        environments = []
+        if self.options.enabled:
+            environments = self.cp.getConsumer(self.identity.uuid)['environments']
+        else:
+            org_environments = self._get_environments(self.org)
+            if self.options.disabled:
+                consumer_id_list = []
+                for env in self.cp.getConsumer(self.identity.uuid)['environments']:
+                    consumer_id_list.append(env['id'])
+                for env in org_environments:
+                    if env['id'] not in consumer_id_list:
+                        environments.append(env)
+            else:
+                environments = org_environments
+
+        if len(environments):
+            print("+-------------------------------------------+")
+            print("          {env}".format(env=_('Environments')))
+            print("+-------------------------------------------+")
+            for env in environments:
+                print(columnize(ENVIRONMENT_LIST, echo_columnize_callback, env['name'],
+                                env['description'] or "") + "\n")
+        else:
+            print(_("This list operation does not have any environments to report."))
+
+    def _process_environments(self, admin_cp, owner_key, options):
+        all_env_list = admin_cp.getEnvironmentList(owner_key)
+        return check_set_environment_names(all_env_list, options.set)
+
+    @property
+    def org(self):
```

Comment:
Why is property of `OrgCommand` parent class overridden here? Is it really necessary? If yes, then explanation in doc string would be desired.